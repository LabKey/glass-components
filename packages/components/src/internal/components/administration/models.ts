interface Member {
    id: number;
    name: string;
    type: string;
}

interface Group {
    groupName: string;
    members: Member[];
}

export interface GroupMembership {
    [key: string]: Group;
}
