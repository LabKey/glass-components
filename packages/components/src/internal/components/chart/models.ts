import { ReactNode } from 'react';
import { Filter } from '@labkey/api';

import { AppURL } from '../../url/AppURL';
import { SchemaQuery } from '../../../public/SchemaQuery';

export interface BarChartData {
    count: number;
    id?: string | number;
    x: string;
    xSub?: string;
}

export interface BarChartSelector {
    filter?: number;
    label: string;
    name: string;
}

export interface BarChartConfig {
    charts: BarChartSelector[];
    colorPath?: string[];
    emptyStateMsg?: ReactNode;
    filterDataRegionName?: string;
    getAppURL?: (data: BarChartData, evt?: any) => AppURL;
    getProjectExclusionFilter?: (projectExclusions: { [key: string]: number[] }) => Filter.IFilter;
    groupPath?: string[];
    itemCountFilters?: Filter.IFilter[];
    itemCountSQ: SchemaQuery;
    key: string;
    label: string;
    namePath?: string[];
    queryName: string;
    schemaName: string;
    showSampleButtons?: boolean;
    sort?: string;
}
