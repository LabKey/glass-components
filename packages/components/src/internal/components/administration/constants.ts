import { PermissionRoles } from '@labkey/api';
import { Map } from 'immutable';

const ROLE_ACTION_DESCRIPTION =
    'and creating and editing sample types, source types, assay designs, and job templates.';

export const SECURITY_ROLE_DESCRIPTIONS = Map<string, string>([
    [
        PermissionRoles.ApplicationAdmin,
        'Application Administrators have full control over the application. This includes user management, permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.ProjectAdmin,
        'Project Administrators have full control over the application in this project container. This includes user management, permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.FolderAdmin,
        'Folder Administrators have control over the folder specific pieces of the application. This includes permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.Editor,
        'Editors may add new information or edit data related to samples, sources, assays, jobs, but not storage locations.',
    ],
    [PermissionRoles.Reader, 'Readers have a read-only view of the application.'],
]);

export const HOSTED_APPLICATION_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.Editor, 'Editor'],
    [PermissionRoles.EditorWithoutDelete, 'Editor without Delete'],
    [PermissionRoles.Reader, 'Reader'],
]);

export const SITE_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.ApplicationAdmin, 'Application Administrator'],
]);

export const APPLICATION_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.ProjectAdmin, 'Project Administrator'],
    [PermissionRoles.FolderAdmin, 'Folder Administrator'],
    [PermissionRoles.Editor, 'Editor'],
    [PermissionRoles.EditorWithoutDelete, 'Editor without Delete'],
    [PermissionRoles.Reader, 'Reader'],
]);


