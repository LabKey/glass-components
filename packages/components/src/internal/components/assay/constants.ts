import { SCHEMAS } from '../../schemas';

export const GENERAL_ASSAY_PROVIDER_NAME = 'General';

export const PLATE_METADATA_COLUMN = 'PlateMetadata';
export const PLATE_TEMPLATE_COLUMN = 'PlateTemplate';

// Columns are required for us to render the WorkflowTask in EditableDetails components
export const WORKFLOW_TASK_PROPERTIES_REQUIRED_COLUMNS = ['WorkflowTask', 'WorkflowTask/Run', 'Protocol/RowId'];

export const RUN_PROPERTIES_REQUIRED_COLUMNS = SCHEMAS.CBMB.concat(
    'Name',
    'RowId',
    'ReplacesRun',
    'ReplacedByRun',
    'DataOutputs',
    'DataOutputs/DataFileUrl',
    'Batch',
    'Folder'
)
    .concat(WORKFLOW_TASK_PROPERTIES_REQUIRED_COLUMNS)
    .toList();
