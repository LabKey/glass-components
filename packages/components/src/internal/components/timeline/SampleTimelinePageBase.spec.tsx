import React from 'react';
import renderer from 'react-test-renderer';

import { App, makeTestActions } from '../../../index';

import SampleTimelineJson from '../../../test/data/SampleTimeline.json';

import { getTestAPIWrapper } from '../../APIWrapper';
import { getSamplesTestAPIWrapper } from '../samples/APIWrapper';

import { TimelineEventModel } from '../auditlog/models';
import { SampleStateType } from '../samples/constants';

import { sleep } from '../../testHelpers';

import { SampleTimelinePageBaseImpl } from './SampleTimelinePageBase';

describe('<SampleTimelinePageBase/>', () => {
    const dummyData = SampleTimelineJson;
    const events: TimelineEventModel[] = [];
    if (dummyData['events']) {
        (dummyData['events'] as []).forEach(event => events.push(TimelineEventModel.create(event, 'UTC')));
    }

    const DEFAULT_PROPS = {
        api: getTestAPIWrapper(jest.fn, {
            samples: getSamplesTestAPIWrapper(jest.fn, {
                getTimelineEvents: () => Promise.resolve(events),
            }),
        }),
        queryModels: {},
        actions: makeTestActions(),
        sampleSet: 'Samples',
        sampleId: 86873,
        sampleName: 'S-20200404-1',
        user: App.TEST_USER_FOLDER_ADMIN,
        skipAuditDetailUserLoading: true,
        timezoneAbbr: 'UTC',
        sampleStatus: {
            label: 'Available for Testing',
            statusType: SampleStateType.Available,
            description: 'Description for testing',
        },
        sampleJobsGidId: 'test',
    };

    const registrationEvent = events[0],
        sampleUpdateEvent = events[2],
        jobCompleted = events[6],
        assayReimportEvent = events[7];

    test('Without selected event', async () => {
        const tree = renderer.create(<SampleTimelinePageBaseImpl {...DEFAULT_PROPS} />);
        await sleep();
        expect(tree).toMatchSnapshot();
    });

    test('With selected sample registration event', async () => {
        const tree = renderer.create(
            <SampleTimelinePageBaseImpl {...DEFAULT_PROPS} initialSelectedEvent={registrationEvent} />
        );
        await sleep();
        expect(tree).toMatchSnapshot();
        tree.unmount();
    });

    test('With selected sample update event', async () => {
        const tree = renderer.create(
            <SampleTimelinePageBaseImpl {...DEFAULT_PROPS} initialSelectedEvent={sampleUpdateEvent} />
        );
        await sleep();
        expect(tree).toMatchSnapshot();
        tree.unmount();
    });

    test('With selected assay re-import event', async () => {
        const tree = renderer.create(
            <SampleTimelinePageBaseImpl {...DEFAULT_PROPS} initialSelectedEvent={assayReimportEvent} />
        );
        await sleep();
        expect(tree).toMatchSnapshot();
        tree.unmount();
    });

    test('With selected job', async () => {
        const tree = renderer.create(
            <SampleTimelinePageBaseImpl {...DEFAULT_PROPS} initialSelectedEvent={jobCompleted} />
        );
        await sleep();
        expect(tree).toMatchSnapshot();
        tree.unmount();
    });
});
