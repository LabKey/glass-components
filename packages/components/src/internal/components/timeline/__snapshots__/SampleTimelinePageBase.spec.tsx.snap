// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleTimelinePageBase/> With selected assay re-import event 1`] = `
<Memo()
  actions={
    Object {
      "addModel": [Function],
      "clearSelections": [Function],
      "loadAllModels": [Function],
      "loadCharts": [Function],
      "loadFirstPage": [Function],
      "loadLastPage": [Function],
      "loadModel": [Function],
      "loadNextPage": [Function],
      "loadPreviousPage": [Function],
      "loadRows": [Function],
      "replaceSelections": [Function],
      "selectAllRows": [Function],
      "selectPage": [Function],
      "selectReport": [Function],
      "selectRow": [Function],
      "setFilters": [Function],
      "setMaxRows": [Function],
      "setOffset": [Function],
      "setSchemaQuery": [Function],
      "setSelections": [Function],
      "setSorts": [Function],
      "setView": [Function],
    }
  }
  api={
    Object {
      "domain": Object {
        "getDomainNamePreviews": [MockFunction],
        "getGenId": [MockFunction],
        "hasExistingDomainData": [MockFunction],
        "setGenId": [MockFunction],
        "validateDomainNameExpressions": [MockFunction],
      },
      "entity": Object {
        "getDataOperationConfirmationData": [MockFunction],
      },
      "picklist": Object {
        "getPicklistCountsBySampleType": [Function],
        "getPicklistDeleteData": [Function],
        "getPicklistFromId": [Function],
      },
      "query": Object {
        "getEntityTypeOptions": [MockFunction],
        "getGridViews": [MockFunction],
        "getQueryDetails": [MockFunction],
        "incrementClientSideMetricCount": [MockFunction],
        "selectDistinctRows": [MockFunction],
        "selectRows": [MockFunction],
      },
      "samples": Object {
        "getFieldLookupFromSelection": [MockFunction],
        "getSampleAliquotRows": [MockFunction],
        "getSampleAssayResultViewConfigs": [MockFunction],
        "getSampleOperationConfirmationData": [MockFunction],
        "getSampleStatuses": [MockFunction],
        "getSampleStorageId": [MockFunction],
        "getSelectionLineageData": [MockFunction],
        "getTimelineEvents": [Function],
        "loadFinderSearches": [MockFunction],
      },
      "security": Object {
        "fetchContainers": [MockFunction],
        "fetchPolicy": [MockFunction],
        "getUserLimitSettings": [MockFunction],
      },
    }
  }
  initialSelectedEvent={
    Immutable.Record {
      "rowId": 1549,
      "eventType": "assays",
      "subEventType": undefined,
      "summary": "Assay Data Re-Import Run",
      "user": Immutable.Map {
        "value": 1005,
        "urlType": "user",
        "displayValue": "Vader",
      },
      "eventUserId": 1005,
      "timestamp": Immutable.Map {
        "value": "2020-04-09 23:57:36.941",
        "formattedValue": "2020-04-09 23:57",
      },
      "eventTimestamp": 2020-04-09T23:57:36.941Z,
      "entity": Immutable.Map {
        "value": Immutable.List [
          "xlsx assay",
          233,
        ],
        "urlType": "assayRun",
        "displayValue": "XLSX Assay",
      },
      "entitySeparator": undefined,
      "metadata": Immutable.List [
        Immutable.Map {
          "field": "Assay",
          "value": Immutable.Map {
            "value": "xlsx assay",
            "urlType": "assays",
            "displayValue": "XLSX Assay",
          },
        },
        Immutable.Map {
          "field": "Assay ID",
          "value": Immutable.Map {
            "value": Immutable.List [
              "xlsx assay",
              233,
            ],
            "urlType": "assayRun",
            "displayValue": "XLSX Assay_2019-10-16_11-54-56",
          },
        },
        Immutable.Map {
          "field": "Completed By",
          "value": Immutable.Map {
            "value": 1005,
            "urlType": "user",
            "displayValue": "Vader",
          },
        },
        Immutable.Map {
          "field": "Date",
          "value": Immutable.Map {
            "value": "2020-04-09 23:57:36.941",
            "formattedValue": "2020-04-09 23:57",
          },
        },
      ],
      "oldData": undefined,
      "newData": undefined,
    }
  }
  queryModels={Object {}}
  sampleId={86873}
  sampleJobsGidId="test"
  sampleName="S-20200404-1"
  sampleSet="Samples"
  sampleStatus={
    Object {
      "description": "Description for testing",
      "label": "Available for Testing",
      "statusType": "Available",
    }
  }
  skipAuditDetailUserLoading={true}
  timezoneAbbr="UTC"
  user={
    Immutable.Record {
      "id": 1005,
      "canDelete": true,
      "canDeleteOwn": true,
      "canInsert": true,
      "canUpdate": true,
      "canUpdateOwn": true,
      "displayName": "FolderAdminDisplayName",
      "email": "guest",
      "phone": null,
      "avatar": "/labkey/_images/defaultavatar.png",
      "isAdmin": true,
      "isAnalyst": true,
      "isDeveloper": true,
      "isGuest": false,
      "isRootAdmin": false,
      "isSignedIn": true,
      "isSystemAdmin": false,
      "isTrusted": true,
      "maxAllowedPhi": undefined,
      "permissionsList": Immutable.List [
        "org.labkey.api.security.permissions.DeletePermission",
        "org.labkey.api.security.permissions.ReadPermission",
        "org.labkey.api.security.permissions.DesignDataClassPermission",
        "org.labkey.api.security.permissions.DesignSampleTypePermission",
        "org.labkey.api.inventory.security.StorageDesignPermission",
        "org.labkey.api.assay.security.DesignAssayPermission",
        "org.labkey.api.security.permissions.InsertPermission",
        "org.labkey.api.security.permissions.UpdatePermission",
        "org.labkey.api.lists.permissions.ManagePicklistsPermission",
        "org.labkey.api.security.permissions.SampleWorkflowJobPermission",
        "org.labkey.api.security.permissions.AdminPermission",
        "org.labkey.api.security.permissions.NotebookReadPermission",
        "org.labkey.api.security.permissions.DataClassReadPermission",
        "org.labkey.api.security.permissions.AssayReadPermission",
        "org.labkey.api.security.permissions.MediaReadPermission",
      ],
    }
  }
>
  <LoadingSpinner
    msg="Loading..."
    wrapperClassName=""
  >
    <span
      className=""
    >
      <i
        aria-hidden="true"
        className="fa fa-spinner fa-pulse"
      />
       
      Loading...
    </span>
  </LoadingSpinner>
</Memo()>
`;

exports[`<SampleTimelinePageBase/> With selected job 1`] = `
<Memo()
  actions={
    Object {
      "addModel": [Function],
      "clearSelections": [Function],
      "loadAllModels": [Function],
      "loadCharts": [Function],
      "loadFirstPage": [Function],
      "loadLastPage": [Function],
      "loadModel": [Function],
      "loadNextPage": [Function],
      "loadPreviousPage": [Function],
      "loadRows": [Function],
      "replaceSelections": [Function],
      "selectAllRows": [Function],
      "selectPage": [Function],
      "selectReport": [Function],
      "selectRow": [Function],
      "setFilters": [Function],
      "setMaxRows": [Function],
      "setOffset": [Function],
      "setSchemaQuery": [Function],
      "setSelections": [Function],
      "setSorts": [Function],
      "setView": [Function],
    }
  }
  api={
    Object {
      "domain": Object {
        "getDomainNamePreviews": [MockFunction],
        "getGenId": [MockFunction],
        "hasExistingDomainData": [MockFunction],
        "setGenId": [MockFunction],
        "validateDomainNameExpressions": [MockFunction],
      },
      "entity": Object {
        "getDataOperationConfirmationData": [MockFunction],
      },
      "picklist": Object {
        "getPicklistCountsBySampleType": [Function],
        "getPicklistDeleteData": [Function],
        "getPicklistFromId": [Function],
      },
      "query": Object {
        "getEntityTypeOptions": [MockFunction],
        "getGridViews": [MockFunction],
        "getQueryDetails": [MockFunction],
        "incrementClientSideMetricCount": [MockFunction],
        "selectDistinctRows": [MockFunction],
        "selectRows": [MockFunction],
      },
      "samples": Object {
        "getFieldLookupFromSelection": [MockFunction],
        "getSampleAliquotRows": [MockFunction],
        "getSampleAssayResultViewConfigs": [MockFunction],
        "getSampleOperationConfirmationData": [MockFunction],
        "getSampleStatuses": [MockFunction],
        "getSampleStorageId": [MockFunction],
        "getSelectionLineageData": [MockFunction],
        "getTimelineEvents": [Function],
        "loadFinderSearches": [MockFunction],
      },
      "security": Object {
        "fetchContainers": [MockFunction],
        "fetchPolicy": [MockFunction],
        "getUserLimitSettings": [MockFunction],
      },
    }
  }
  initialSelectedEvent={
    Immutable.Record {
      "rowId": 548,
      "eventType": "workflow",
      "subEventType": undefined,
      "summary": "Completed job",
      "user": Immutable.Map {
        "value": 1005,
        "urlType": "user",
        "displayValue": "Vader",
      },
      "eventUserId": 1005,
      "timestamp": Immutable.Map {
        "value": "2020-04-09 20:57:36.941",
        "formattedValue": "2020-04-09 20:57",
      },
      "eventTimestamp": 2020-04-09T20:57:36.941Z,
      "entity": Immutable.Map {
        "value": 687,
        "urlType": "workflow",
        "displayValue": "job-1",
      },
      "entitySeparator": undefined,
      "metadata": Immutable.List [
        Immutable.Map {
          "field": "Job Name",
          "value": Immutable.Map {
            "value": 687,
            "urlType": "workflow",
            "displayValue": "job-1",
          },
        },
        Immutable.Map {
          "field": "Job Owner",
          "value": Immutable.Map {
            "value": 1005,
            "urlType": "user",
            "displayValue": "Vader",
          },
        },
        Immutable.Map {
          "field": "Job Status",
          "value": "Completed",
        },
      ],
      "oldData": undefined,
      "newData": undefined,
    }
  }
  queryModels={Object {}}
  sampleId={86873}
  sampleJobsGidId="test"
  sampleName="S-20200404-1"
  sampleSet="Samples"
  sampleStatus={
    Object {
      "description": "Description for testing",
      "label": "Available for Testing",
      "statusType": "Available",
    }
  }
  skipAuditDetailUserLoading={true}
  timezoneAbbr="UTC"
  user={
    Immutable.Record {
      "id": 1005,
      "canDelete": true,
      "canDeleteOwn": true,
      "canInsert": true,
      "canUpdate": true,
      "canUpdateOwn": true,
      "displayName": "FolderAdminDisplayName",
      "email": "guest",
      "phone": null,
      "avatar": "/labkey/_images/defaultavatar.png",
      "isAdmin": true,
      "isAnalyst": true,
      "isDeveloper": true,
      "isGuest": false,
      "isRootAdmin": false,
      "isSignedIn": true,
      "isSystemAdmin": false,
      "isTrusted": true,
      "maxAllowedPhi": undefined,
      "permissionsList": Immutable.List [
        "org.labkey.api.security.permissions.DeletePermission",
        "org.labkey.api.security.permissions.ReadPermission",
        "org.labkey.api.security.permissions.DesignDataClassPermission",
        "org.labkey.api.security.permissions.DesignSampleTypePermission",
        "org.labkey.api.inventory.security.StorageDesignPermission",
        "org.labkey.api.assay.security.DesignAssayPermission",
        "org.labkey.api.security.permissions.InsertPermission",
        "org.labkey.api.security.permissions.UpdatePermission",
        "org.labkey.api.lists.permissions.ManagePicklistsPermission",
        "org.labkey.api.security.permissions.SampleWorkflowJobPermission",
        "org.labkey.api.security.permissions.AdminPermission",
        "org.labkey.api.security.permissions.NotebookReadPermission",
        "org.labkey.api.security.permissions.DataClassReadPermission",
        "org.labkey.api.security.permissions.AssayReadPermission",
        "org.labkey.api.security.permissions.MediaReadPermission",
      ],
    }
  }
>
  <LoadingSpinner
    msg="Loading..."
    wrapperClassName=""
  >
    <span
      className=""
    >
      <i
        aria-hidden="true"
        className="fa fa-spinner fa-pulse"
      />
       
      Loading...
    </span>
  </LoadingSpinner>
</Memo()>
`;

exports[`<SampleTimelinePageBase/> With selected sample registration event 1`] = `
<Memo()
  actions={
    Object {
      "addModel": [Function],
      "clearSelections": [Function],
      "loadAllModels": [Function],
      "loadCharts": [Function],
      "loadFirstPage": [Function],
      "loadLastPage": [Function],
      "loadModel": [Function],
      "loadNextPage": [Function],
      "loadPreviousPage": [Function],
      "loadRows": [Function],
      "replaceSelections": [Function],
      "selectAllRows": [Function],
      "selectPage": [Function],
      "selectReport": [Function],
      "selectRow": [Function],
      "setFilters": [Function],
      "setMaxRows": [Function],
      "setOffset": [Function],
      "setSchemaQuery": [Function],
      "setSelections": [Function],
      "setSorts": [Function],
      "setView": [Function],
    }
  }
  api={
    Object {
      "domain": Object {
        "getDomainNamePreviews": [MockFunction],
        "getGenId": [MockFunction],
        "hasExistingDomainData": [MockFunction],
        "setGenId": [MockFunction],
        "validateDomainNameExpressions": [MockFunction],
      },
      "entity": Object {
        "getDataOperationConfirmationData": [MockFunction],
      },
      "picklist": Object {
        "getPicklistCountsBySampleType": [Function],
        "getPicklistDeleteData": [Function],
        "getPicklistFromId": [Function],
      },
      "query": Object {
        "getEntityTypeOptions": [MockFunction],
        "getGridViews": [MockFunction],
        "getQueryDetails": [MockFunction],
        "incrementClientSideMetricCount": [MockFunction],
        "selectDistinctRows": [MockFunction],
        "selectRows": [MockFunction],
      },
      "samples": Object {
        "getFieldLookupFromSelection": [MockFunction],
        "getSampleAliquotRows": [MockFunction],
        "getSampleAssayResultViewConfigs": [MockFunction],
        "getSampleOperationConfirmationData": [MockFunction],
        "getSampleStatuses": [MockFunction],
        "getSampleStorageId": [MockFunction],
        "getSelectionLineageData": [MockFunction],
        "getTimelineEvents": [Function],
        "loadFinderSearches": [MockFunction],
      },
      "security": Object {
        "fetchContainers": [MockFunction],
        "fetchPolicy": [MockFunction],
        "getUserLimitSettings": [MockFunction],
      },
    }
  }
  queryModels={Object {}}
  sampleId={86873}
  sampleJobsGidId="test"
  sampleName="S-20200404-1"
  sampleSet="Samples"
  sampleStatus={
    Object {
      "description": "Description for testing",
      "label": "Available for Testing",
      "statusType": "Available",
    }
  }
  skipAuditDetailUserLoading={true}
  timezoneAbbr="UTC"
  user={
    Immutable.Record {
      "id": 1005,
      "canDelete": true,
      "canDeleteOwn": true,
      "canInsert": true,
      "canUpdate": true,
      "canUpdateOwn": true,
      "displayName": "FolderAdminDisplayName",
      "email": "guest",
      "phone": null,
      "avatar": "/labkey/_images/defaultavatar.png",
      "isAdmin": true,
      "isAnalyst": true,
      "isDeveloper": true,
      "isGuest": false,
      "isRootAdmin": false,
      "isSignedIn": true,
      "isSystemAdmin": false,
      "isTrusted": true,
      "maxAllowedPhi": undefined,
      "permissionsList": Immutable.List [
        "org.labkey.api.security.permissions.DeletePermission",
        "org.labkey.api.security.permissions.ReadPermission",
        "org.labkey.api.security.permissions.DesignDataClassPermission",
        "org.labkey.api.security.permissions.DesignSampleTypePermission",
        "org.labkey.api.inventory.security.StorageDesignPermission",
        "org.labkey.api.assay.security.DesignAssayPermission",
        "org.labkey.api.security.permissions.InsertPermission",
        "org.labkey.api.security.permissions.UpdatePermission",
        "org.labkey.api.lists.permissions.ManagePicklistsPermission",
        "org.labkey.api.security.permissions.SampleWorkflowJobPermission",
        "org.labkey.api.security.permissions.AdminPermission",
        "org.labkey.api.security.permissions.NotebookReadPermission",
        "org.labkey.api.security.permissions.DataClassReadPermission",
        "org.labkey.api.security.permissions.AssayReadPermission",
        "org.labkey.api.security.permissions.MediaReadPermission",
      ],
    }
  }
>
  <LoadingSpinner
    msg="Loading..."
    wrapperClassName=""
  >
    <span
      className=""
    >
      <i
        aria-hidden="true"
        className="fa fa-spinner fa-pulse"
      />
       
      Loading...
    </span>
  </LoadingSpinner>
</Memo()>
`;

exports[`<SampleTimelinePageBase/> With selected sample update event 1`] = `
<Memo()
  actions={
    Object {
      "addModel": [Function],
      "clearSelections": [Function],
      "loadAllModels": [Function],
      "loadCharts": [Function],
      "loadFirstPage": [Function],
      "loadLastPage": [Function],
      "loadModel": [Function],
      "loadNextPage": [Function],
      "loadPreviousPage": [Function],
      "loadRows": [Function],
      "replaceSelections": [Function],
      "selectAllRows": [Function],
      "selectPage": [Function],
      "selectReport": [Function],
      "selectRow": [Function],
      "setFilters": [Function],
      "setMaxRows": [Function],
      "setOffset": [Function],
      "setSchemaQuery": [Function],
      "setSelections": [Function],
      "setSorts": [Function],
      "setView": [Function],
    }
  }
  api={
    Object {
      "domain": Object {
        "getDomainNamePreviews": [MockFunction],
        "getGenId": [MockFunction],
        "hasExistingDomainData": [MockFunction],
        "setGenId": [MockFunction],
        "validateDomainNameExpressions": [MockFunction],
      },
      "entity": Object {
        "getDataOperationConfirmationData": [MockFunction],
      },
      "picklist": Object {
        "getPicklistCountsBySampleType": [Function],
        "getPicklistDeleteData": [Function],
        "getPicklistFromId": [Function],
      },
      "query": Object {
        "getEntityTypeOptions": [MockFunction],
        "getGridViews": [MockFunction],
        "getQueryDetails": [MockFunction],
        "incrementClientSideMetricCount": [MockFunction],
        "selectDistinctRows": [MockFunction],
        "selectRows": [MockFunction],
      },
      "samples": Object {
        "getFieldLookupFromSelection": [MockFunction],
        "getSampleAliquotRows": [MockFunction],
        "getSampleAssayResultViewConfigs": [MockFunction],
        "getSampleOperationConfirmationData": [MockFunction],
        "getSampleStatuses": [MockFunction],
        "getSampleStorageId": [MockFunction],
        "getSelectionLineageData": [MockFunction],
        "getTimelineEvents": [Function],
        "loadFinderSearches": [MockFunction],
      },
      "security": Object {
        "fetchContainers": [MockFunction],
        "fetchPolicy": [MockFunction],
        "getUserLimitSettings": [MockFunction],
      },
    }
  }
  initialSelectedEvent={
    Immutable.Record {
      "rowId": 462023,
      "eventType": "samples",
      "subEventType": undefined,
      "summary": "Sample updated",
      "user": Immutable.Map {
        "value": 1005,
        "urlType": "user",
        "displayValue": "Vader",
      },
      "eventUserId": 1005,
      "timestamp": Immutable.Map {
        "value": "2020-04-05 21:57:36.941",
        "formattedValue": "2020-04-05 21:57",
      },
      "eventTimestamp": 2020-04-05T21:57:36.941Z,
      "entity": undefined,
      "entitySeparator": undefined,
      "metadata": undefined,
      "oldData": Immutable.Map {
        "description": "des",
        "modified": "2020-04-05T21:56:54.584-07:00",
      },
      "newData": Immutable.Map {
        "description": "desc",
        "modified": "2020-04-05T21:57:36.915-07:00",
      },
    }
  }
  queryModels={Object {}}
  sampleId={86873}
  sampleJobsGidId="test"
  sampleName="S-20200404-1"
  sampleSet="Samples"
  sampleStatus={
    Object {
      "description": "Description for testing",
      "label": "Available for Testing",
      "statusType": "Available",
    }
  }
  skipAuditDetailUserLoading={true}
  timezoneAbbr="UTC"
  user={
    Immutable.Record {
      "id": 1005,
      "canDelete": true,
      "canDeleteOwn": true,
      "canInsert": true,
      "canUpdate": true,
      "canUpdateOwn": true,
      "displayName": "FolderAdminDisplayName",
      "email": "guest",
      "phone": null,
      "avatar": "/labkey/_images/defaultavatar.png",
      "isAdmin": true,
      "isAnalyst": true,
      "isDeveloper": true,
      "isGuest": false,
      "isRootAdmin": false,
      "isSignedIn": true,
      "isSystemAdmin": false,
      "isTrusted": true,
      "maxAllowedPhi": undefined,
      "permissionsList": Immutable.List [
        "org.labkey.api.security.permissions.DeletePermission",
        "org.labkey.api.security.permissions.ReadPermission",
        "org.labkey.api.security.permissions.DesignDataClassPermission",
        "org.labkey.api.security.permissions.DesignSampleTypePermission",
        "org.labkey.api.inventory.security.StorageDesignPermission",
        "org.labkey.api.assay.security.DesignAssayPermission",
        "org.labkey.api.security.permissions.InsertPermission",
        "org.labkey.api.security.permissions.UpdatePermission",
        "org.labkey.api.lists.permissions.ManagePicklistsPermission",
        "org.labkey.api.security.permissions.SampleWorkflowJobPermission",
        "org.labkey.api.security.permissions.AdminPermission",
        "org.labkey.api.security.permissions.NotebookReadPermission",
        "org.labkey.api.security.permissions.DataClassReadPermission",
        "org.labkey.api.security.permissions.AssayReadPermission",
        "org.labkey.api.security.permissions.MediaReadPermission",
      ],
    }
  }
>
  <LoadingSpinner
    msg="Loading..."
    wrapperClassName=""
  >
    <span
      className=""
    >
      <i
        aria-hidden="true"
        className="fa fa-spinner fa-pulse"
      />
       
      Loading...
    </span>
  </LoadingSpinner>
</Memo()>
`;

exports[`<SampleTimelinePageBase/> Without selected event 1`] = `
<Memo()
  actions={
    Object {
      "addModel": [Function],
      "clearSelections": [Function],
      "loadAllModels": [Function],
      "loadCharts": [Function],
      "loadFirstPage": [Function],
      "loadLastPage": [Function],
      "loadModel": [Function],
      "loadNextPage": [Function],
      "loadPreviousPage": [Function],
      "loadRows": [Function],
      "replaceSelections": [Function],
      "selectAllRows": [Function],
      "selectPage": [Function],
      "selectReport": [Function],
      "selectRow": [Function],
      "setFilters": [Function],
      "setMaxRows": [Function],
      "setOffset": [Function],
      "setSchemaQuery": [Function],
      "setSelections": [Function],
      "setSorts": [Function],
      "setView": [Function],
    }
  }
  api={
    Object {
      "domain": Object {
        "getDomainNamePreviews": [MockFunction],
        "getGenId": [MockFunction],
        "hasExistingDomainData": [MockFunction],
        "setGenId": [MockFunction],
        "validateDomainNameExpressions": [MockFunction],
      },
      "entity": Object {
        "getDataOperationConfirmationData": [MockFunction],
      },
      "picklist": Object {
        "getPicklistCountsBySampleType": [Function],
        "getPicklistDeleteData": [Function],
        "getPicklistFromId": [Function],
      },
      "query": Object {
        "getEntityTypeOptions": [MockFunction],
        "getGridViews": [MockFunction],
        "getQueryDetails": [MockFunction],
        "incrementClientSideMetricCount": [MockFunction],
        "selectDistinctRows": [MockFunction],
        "selectRows": [MockFunction],
      },
      "samples": Object {
        "getFieldLookupFromSelection": [MockFunction],
        "getSampleAliquotRows": [MockFunction],
        "getSampleAssayResultViewConfigs": [MockFunction],
        "getSampleOperationConfirmationData": [MockFunction],
        "getSampleStatuses": [MockFunction],
        "getSampleStorageId": [MockFunction],
        "getSelectionLineageData": [MockFunction],
        "getTimelineEvents": [Function],
        "loadFinderSearches": [MockFunction],
      },
      "security": Object {
        "fetchContainers": [MockFunction],
        "fetchPolicy": [MockFunction],
        "getUserLimitSettings": [MockFunction],
      },
    }
  }
  queryModels={Object {}}
  sampleId={86873}
  sampleJobsGidId="test"
  sampleName="S-20200404-1"
  sampleSet="Samples"
  sampleStatus={
    Object {
      "description": "Description for testing",
      "label": "Available for Testing",
      "statusType": "Available",
    }
  }
  skipAuditDetailUserLoading={true}
  timezoneAbbr="UTC"
  user={
    Immutable.Record {
      "id": 1005,
      "canDelete": true,
      "canDeleteOwn": true,
      "canInsert": true,
      "canUpdate": true,
      "canUpdateOwn": true,
      "displayName": "FolderAdminDisplayName",
      "email": "guest",
      "phone": null,
      "avatar": "/labkey/_images/defaultavatar.png",
      "isAdmin": true,
      "isAnalyst": true,
      "isDeveloper": true,
      "isGuest": false,
      "isRootAdmin": false,
      "isSignedIn": true,
      "isSystemAdmin": false,
      "isTrusted": true,
      "maxAllowedPhi": undefined,
      "permissionsList": Immutable.List [
        "org.labkey.api.security.permissions.DeletePermission",
        "org.labkey.api.security.permissions.ReadPermission",
        "org.labkey.api.security.permissions.DesignDataClassPermission",
        "org.labkey.api.security.permissions.DesignSampleTypePermission",
        "org.labkey.api.inventory.security.StorageDesignPermission",
        "org.labkey.api.assay.security.DesignAssayPermission",
        "org.labkey.api.security.permissions.InsertPermission",
        "org.labkey.api.security.permissions.UpdatePermission",
        "org.labkey.api.lists.permissions.ManagePicklistsPermission",
        "org.labkey.api.security.permissions.SampleWorkflowJobPermission",
        "org.labkey.api.security.permissions.AdminPermission",
        "org.labkey.api.security.permissions.NotebookReadPermission",
        "org.labkey.api.security.permissions.DataClassReadPermission",
        "org.labkey.api.security.permissions.AssayReadPermission",
        "org.labkey.api.security.permissions.MediaReadPermission",
      ],
    }
  }
>
  <LoadingSpinner
    msg="Loading..."
    wrapperClassName=""
  >
    <span
      className=""
    >
      <i
        aria-hidden="true"
        className="fa fa-spinner fa-pulse"
      />
       
      Loading...
    </span>
  </LoadingSpinner>
</Memo()>
`;
